Sources/AngouriMath/Functions/Series.cs

```diff
@@ -44,5 +45,110 @@ internal static Entity TaylorExpansion(Entity expr, Variable exprVariable, Varia
+                            {
+                                var newTerm = newTerms[newTermIndex];
+
+                                if (Enumerable.SequenceEqual(newPointCoeffs, newTerm.pointCoefficientDegrees))
+                                {
+                                    // If the match is found, instead of creating a whole new repeat term, we just add
+                                    // to the match's coefficient. This ends up making a neat binomial coefficient pattern.
+                                    newTerm.coefficient += lastTerm.coefficient;
+                                    foundRepeatFlag = true;
+                                    break;
+                                }
+                            }
+
+                            // If no match is found, we add it to the new term list.
+                            if (!foundRepeatFlag)
+                            {
+                                var newPartialDerivative = lastTerm.partialDerivative.Differentiate(variable.exprVariable);
+                                newTerms.Add((lastTerm.coefficient, newPointCoeffs, newPartialDerivative));
+                            }
+                           
+                        }
+                    }
+
+                }
+
+
+                // From the list of deconstructed terms, we put assemble the term and sum them all up.
+                var fullExpressionTerms = new List<Entity>();
+                foreach (var newTerm in newTerms) 
+                {
+                    // pointCoefficients are transformed into their proper (x - a)^t... forms.
+                    Entity pointCoefficients = 1;
+                    for (int variableIndex = 0; variableIndex < exprToPolyVars.Length; variableIndex++)
+                    {
+                        var triple = exprToPolyVars[variableIndex];
+                        pointCoefficients *= (triple.polyVariable - triple.point).Pow(newTerm.pointCoefficientDegrees[variableIndex]);
+                    }
+
+                    fullExpressionTerms.Add( newTerm.partialDerivative * newTerm.coefficient * pointCoefficients );
+                }
+                Entity fullExpr = TreeAnalyzer.MultiHangBinary(fullExpressionTerms, (a, b) => a + b);
+
+                // then solve for the given point and simplify,
+                foreach (var variable in exprToPolyVars)
+                    fullExpr = fullExpr.Substitute(variable.exprVariable, variable.point).InnerSimplified;
+
+                // and divide that all by the proper factorial.
+                if (order > 1)
+                    fullExpr /= ((Entity)order).Factorial();
+
+                yield return fullExpr;
+
+                lastTerms = newTerms;
+                order++;
+            }
+        }
+
+        internal static Entity MultivariableTaylorExpansion(Entity expr, (Variable exprVariable, Variable polyVariable, Entity point)[] exprToPolyVars, int termCount)
```

Comment:
Also, could you slightly change the API? To be precise, move `termCount` after `expr` and mark the array with `params` (so that we don't have to write a lot of unnecessary code, you know)